### DISCUSSION TITLE
#(widget) Let your trilium integrate drawio

### DISCUSSION THREAD

**Discussion by SiriusXT** [TriliumNext/Trilium#3891] [Show and tell]
## Installation
1. Create a code note of type JS Frontend with the contents of `trilium-drawio.js` and the label `#widget`
2. Import `drawio.svg` into Notes and add `#template` tag to it
3. Reload frontend
https://github.com/SiriusXT/trilium-drawio
## Tips
1. The saved format is svg
2. Click on the svg image to enter the editing state
3. Will call the drawio website

![preview1](https://user-images.githubusercontent.com/37627919/235350924-55987a7e-91b1-4f9c-8fac-6ce42ace765d.gif)

**Comment by meichthys** [comment]
This is awesome!
One thing i noticed is that after saving the svg is shown, but if you browser away from the draw.io node and then back to it, the svg is not shown until you click on the blank note body to open the editor. Is this expected?

EDIT: nevermind - you're on top of it - this has been [reported and fixed](https://github.com/SiriusXT/trilium-drawio/issues/1) 👍

**Comment by meichthys** [comment]
Keep these plugins coming!

**Comment by Nriver** [comment]
That's so cool!

**Comment by nema69** [comment]
This is awesome!

**Comment by newmedicine** [comment]
Thank you for this, I love draw.io and was thrilled when I found this plugin.

Is it just me or does the shapes panel on the left not display (as it would on diagrams.net)? Using version:0.3.2 and Trilium 0.60.4

<img width="612" alt="image" src="https://github.com/zadam/trilium/assets/22443747/33c4c311-acbe-48c7-847b-5c9f49194e4a">
